lib/posix/spawn/child.rb

```diff
@@ -148,8 +154,9 @@ def exec!
       rescue Object => boom
         [stdin, stdout, stderr].each { |fd| fd.close rescue nil }
         if @status.nil?
-          ::Process.kill('TERM', pid) rescue nil
-          @status = waitpid(pid)      rescue nil
+          kill_pid = @pgroup_kill ? pid * -1 : pid
```

Comment:
This feels a tad clever to me. Maybe just repeat the `Process.kill` lines to make it clear something different is happening:

```ruby 
if @pgroup_kill
  # Negative PID kills the process group with that id
  Process.kill("TERM", -(pid))
else
  Process.kill("TERM", pid)
end
```

Also, fwiw, the docs for `Process.kill` mention negating the signal, not the pid, to kill a group. Not sure which is more idiomatic. 

> [If signal is negative (or starts with a minus sign), kills process groups instead of processes.](http://ruby-doc.org/core-1.9.3/Process.html#method-c-kill)